nbdev/cli.py

```diff
@@ -82,13 +82,30 @@ def _render_nb(fn, cfg):
     fn.write_text(txt)
 
 # %% ../nbs/09_API/12_cli.ipynb 11
+def _update_repo_meta(cfg):
+    "Enable gh pages and update the homepage and description in your GitHub repo."
+    token=os.getenv('GITHUB_TOKEN')
+    if token: 
+        from ghapi.core import GhApi
+        api = GhApi(owner=cfg.user, repo=cfg.repo, token=token)
+        try: # repo needs something in it before you can enable pages
+            cmds = L(['git add settings.ini', "git commit -m'add settings'", 'git config push.default current', 'git push']) 
```

Comment:
When you are first initializing a github repo, it cannot be empty before you enable pages.  It has to contain at least one file.  I don't love executing git commands on behalf of the user.  Another alternative would be to commit an empty file, but that clutters the repo IMO.  

Curious on what your opinions are here.  I put this in its own try block because I suspect something could go wrong here and I don't want to disrupt the onboarding process.

Reply:
I think it's fine - whether we exec git for the user vs use ghapi for the user seems much the same afaict.

FYI I don't think `map` here is useful -- seems a plain `for` loop would be simpler and just as concise. Generally `map` is most useful for the situation where you want to return a list.

Reply:
At least until we release it, perhaps best *not* to have the `try`, so we can find out if there are errors, and if so, what errors come up?